### ISSUE TITLE
Does this method apply to human pose estimation?

### ISSUE THREAD

**Issue by SuperheadNick** [zxz267/HaMuCo#1]
Thanks for your excellent work! Here I have one question to ask: 
Does this method apply to human pose estimation? I tried to train the model with H36M dataset, but severe overfitting seems to occur.

**Comment by zxz267** [comment]
Thanks for your interest in our work. We attempted to employ our approach for human pose estimation and achieved comparable performance to EpipolarPose and CanonPose. Have you experimented with integrating the body prior (e.g., Vposer) for regularization? As far as I recall, it's one of the main adjustments when applying HaMuCo to human pose estimation.

**Comment by SuperheadNick** [comment]
Thanks for your answer. I tried to integrate Vposer for regularization as you said. However, I trained the whole framework for 10 epochs and got the result like this (the former is the training result and the latter is the validation result): 
![1703734158060](https://github.com/zxz267/HaMuCo/assets/74962034/88a83062-0d06-480d-adea-9791e3eba114)
![1703734182567](https://github.com/zxz267/HaMuCo/assets/74962034/d95e1892-cfb5-4d81-8a2d-aec8667b1af2)
It seems overfitting happens again (especially pose_xyz_0), Have you ever experienced this? Does this have anything to do with the weight setting of my regularized loss function, currently I still set it to 0.01 as given in your code.
Additionally, I previously trained a version that did not utilize vposer but only smpl for regularization, I trained it for 30 epochs, and the results were as follows.
![1703733579690](https://github.com/zxz267/HaMuCo/assets/74962034/60c18f0b-87f6-4a1a-9dcc-5aae82b1745b)
![1703733594847](https://github.com/zxz267/HaMuCo/assets/74962034/79b1907a-ea66-492a-9401-49872f89366e)

**Comment by zxz267** [comment]
Due to the limited number of views on H36M and the discrepancies between the body and hands, there may be a potential issue of overfitting without further adjustments. Hence, it would be advisable to experiment with various weight settings. By the way, could you please elaborate on how you integrate VPoser into the framework? Furthermore, which off-the-shelf estimator are you employing? If memory serves me correctly, utilizing AlphaPose yielded superior performance in comparison to OpenPose.

**Comment by zxz267** [comment]
> Thanks for your answer. I tried to integrate Vposer for regularization as you said. However, I trained the whole framework for 10 epochs and got the result like this (the former is the training result and the latter is the validation result): ![1703734158060](https://private-user-images.githubusercontent.com/74962034/293131785-88a83062-0d06-480d-adea-9791e3eba114.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyNzAzMDIsIm5iZiI6MTcwNDI3MDAwMiwicGF0aCI6Ii83NDk2MjAzNC8yOTMxMzE3ODUtODhhODMwNjItMGQwNi00ODBkLWFkZWEtOTc5MWUzZWJhMTE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAzVDA4MjAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgzZGZiNWFmZGIyZWJhMGY1YTk1YjNiNDlkNGFjOTdkZjRkMmFmZTgxODhmM2Q5ZjRiMGY2OGM1MGZkMmFmNjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.g9YhFwaKeswZSNd7SQ-GiNAAUa8zNRvZebt_rMUGRvQ) ![1703734182567](https://private-user-images.githubusercontent.com/74962034/293131796-d95e1892-cfb5-4d81-8a2d-aec8667b1af2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyNzAzMDIsIm5iZiI6MTcwNDI3MDAwMiwicGF0aCI6Ii83NDk2MjAzNC8yOTMxMzE3OTYtZDk1ZTE4OTItY2ZiNS00ZDgxLThhMmQtYWVjODY2N2IxYWYyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAzVDA4MjAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQwNGVhOTQ1MThhMDRjYjUxNjE1ODZmNjczYjRhYTVlZWQwZGUzZmVmYzAyN2NjMDgxY2MzZTYyNDFiZWY5MDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.yQ_n82AAgOsz5PIsLSw8027fCAh9SpWfWHkCYB214us) It seems overfitting happens again (especially pose_xyz_0), Have you ever experienced this? Does this have anything to do with the weight setting of my regularized loss function, currently I still set it to 0.01 as given in your code. Additionally, I previously trained a version that did not utilize vposer but only smpl for regularization, I trained it for 30 epochs, and the results were as follows. ![1703733579690](https://private-user-images.githubusercontent.com/74962034/293131277-60c18f0b-87f6-4a1a-9dcc-5aae82b1745b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyNzAzMDIsIm5iZiI6MTcwNDI3MDAwMiwicGF0aCI6Ii83NDk2MjAzNC8yOTMxMzEyNzctNjBjMThmMGItODdmNi00YTFhLTlkY2MtNWFhZTgyYjE3NDViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAzVDA4MjAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMxOTc2NmU1NjRjMWNjNzIyOTNmOTc3ZjgxNTRmZGM5YmNlZDNiNzA5OGU4OTI5ZGNhYzIxOTRiZWQxYmFlNzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.azQgnt_2E-QfeqbwhfLwWQbxqv32Vd97H9p4Z7Dr2uY) ![1703733594847](https://private-user-images.githubusercontent.com/74962034/293131286-79b1907a-ea66-492a-9401-49872f89366e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyNzAzMDIsIm5iZiI6MTcwNDI3MDAwMiwicGF0aCI6Ii83NDk2MjAzNC8yOTMxMzEyODYtNzliMTkwN2EtZWE2Ni00OTJhLTk0MDEtNDk4NzJmODkzNjZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAzVDA4MjAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgyMjA0Y2ZkMDZlYjQxMDY5MmYwMDI0YTA5ZWVkOWEwNjMyNzBkZjc5NTljODAzODk1YzJmZGI0ZGFlMTQ4ODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.etiHX_sKYMyQhRSCHLBDsD6xyEz9Ercq2-GLYMl2tGk)

Due to the limited number of views on H36M and the discrepancies between the body and hands, there may be a potential issue of overfitting without further adjustments. Hence, it would be advisable to experiment with various weight settings. By the way, could you please elaborate on how you integrate VPoser into the framework? Furthermore, which off-the-shelf estimator are you employing? If memory serves me correctly, utilizing AlphaPose yielded superior performance in comparison to OpenPose.

**Comment by SuperheadNick** [comment]
My current approach is to feed the model-predicted pose to the encoder of a pre-trained Vposer model and use L1loss to constrain the distribution of the pose in this latent space to conform to a Gaussian distribution, but doing so doesn't seem to work as well as expected. May I ask if this is how you were using vposer? Also, I am currently using the 2D ground truth given in the Human3.6M dataset directly, and have not used any other 2D pose estimators as of yet.

**Comment by zxz267** [comment]
> My current approach is to feed the model-predicted pose to the encoder of a pre-trained Vposer model and use L1loss to constrain the distribution of the pose in this latent space to conform to a Gaussian distribution, but doing so doesn't seem to work as well as expected. May I ask if this is how you were using vposer? Also, I am currently using the 2D ground truth given in the Human3.6M dataset directly, and have not used any other 2D pose estimators as of yet.

I used the Vposer as the decoder (modify the network to estimate the latent code of VPoser), similar to [1]. Also, I employed the regularization, as mentioned in [1].
[1] Moon, Gyeongsik, et al. "Three Recipes for Better 3D Pseudo-GTs of 3D Human Mesh Estimation in the Wild." Proceedings of the IEEE/CVF Conference on Computer Vision and Pattern Recognition. 2023.
